src/etc/inc/system.inc

```diff
@@ -1603,7 +1603,6 @@ function system_ntp_setup_gps($serialport) {
 		return false;
 	}
 
-	init_config_arr(array('ntpd', 'gps', 'speed'));
```

Comment:
Why remove this?

Reply:
This is incorrect:
there is no array of speeds in config.xml,
and if it's set as an array, this code is not working:
```php
if (!empty($config['ntpd']['gps']['speed']) && array_key_exists($config['ntpd']['gps']['speed'], $speeds)) {
                $gpsbaud = $speeds[$config['ntpd']['gps']['speed']];
}
```
that's why the speed is always set to 4800

Reply:
Then it should still initialize `ntpd` and `gps` before use.

Reply:
ok, done
mwexec return code checking also fixed -
grep returns zero if it found something vs. if it hasn't found anything (non-zero)